### PR TITLE
Hp 46010A converter

### PR DESCRIPTION
**Pull request by listofoptions**
Converter for the eponymous keyboard.
Thanks for QMK btw! there should probably be a guide for making a converter, as it was a bugbear to figure out that i needed to make a custom matrix.c. i ended up using the code from the m0110 converter as a starting point.

a shallow merge (dont remember if thats the correct term?) might be a good idea as its been a bit since Ive used git. also as c45a41c  is the latest commit that actually works

### CODE REVIEW

File: keyboards/converter/HP_46010A/config.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
```
Comment (drashna) [inline keyboards/converter/HP_46010A/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/converter/HP_46010A/masks.csv [hunk 2]
```diff
@@ -0,0 +1,112 @@
+menu,1,8,1
```
Comment (drashna) [inline keyboards/converter/HP_46010A/masks.csv]: @jackhumbert or @skullydazed Is this file okay, or do we want to remove it?

File: keyboards/converter/HP_46010A/readme.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+Hardware Supported: HP 46010A, Teensy 2.0  
+
+Make example for this keyboard (after setting up your build environment):
+
+    make converter/HP_46010A:default
+
+included in this directory is a arduino program for teensy 2.0 (only one tested, might work on others) to grab a copy of the keyboard matrix.
+also included is a csv containing the matrix as an ordered by clock #, byte array
+    
+See [build environment setup](https://docs.qmk.fm/build_environment_setup.html) then the [make instructions](https://docs.qmk.fm/make_instructions.html) for more information.
```
Comment (drashna) [inline keyboards/converter/HP_46010A/readme.md]: Could you replace this link block with this, instead: 

```markdown
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/converter/HP_46010A/rules.mk [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+# Bootloader
+#     This definition is optional, and if your keyboard supports multiple bootloaders of
+#     different sizes, comment this out, and the correct address will be loaded 
+#     automatically (+60). See bootloader.mk for all options.
+#BOOTLOADER = halfkay
```
Comment (drashna) [inline keyboards/converter/HP_46010A/rules.mk]: This should actually be defined, or you may have issues with booting into the bootloader, and the size calculation during compile will be wrong.

File: keyboards/converter/HP_46010A/rules.mk [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+API_SYSEX_ENABLE = no
+SPLIT_KEYBOARD = no
+WAIT_FOR_USB = yes
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
+LAYOUTS = HP_46010A
```
Comment (drashna) [inline keyboards/converter/HP_46010A/rules.mk]: Could you remove this line?  

It doesn't match any known layouts, so no reason to have it here.

File: keyboards/converter/HP_46010A/masks.csv (hunk 2 above)
Comment (listofoptions) [reply keyboards/converter/HP_46010A/masks.csv]: i just added this for documentation tbh. Im hoping other HP 46010A boards are not different, as hp might have had different switch layouts, therefore the bit mask generated by the arduino program might be different.

File: keyboards/converter/HP_46010A/config.h (hunk 1 above)
Comment (listofoptions) [reply keyboards/converter/HP_46010A/config.h]: sure didnt know that was usable with gcc, thought that was a MSVC only kind of deal

File: keyboards/converter/HP_46010A/readme.md (hunk 3 above)
Comment (listofoptions) [reply keyboards/converter/HP_46010A/readme.md]: ok so i assume you also want me to remove the arduino sketch too?

File: keyboards/converter/HP_46010A/rules.mk (hunk 5 above)
Comment (listofoptions) [reply keyboards/converter/HP_46010A/rules.mk]: ah, i thought it was needed for compiling the layout. need to read the make files better next time

File: keyboards/converter/hp_46010a/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+     * |------------------------------------------------------------------------||---------|    |-------------------|
+     * |caps|ctrl|   a|   s|   d|   f|   g|   h|   j|   k|   l|   ;|   '|   retr||xxxx|pgup|    |   4|   5|   6|   ,|
+     * |------------------------------------------------------------------------------------    |-------------------|
+     * |del |lsft  |   z|   x|   c|   v|   b|   n|   m|   ,|   .|   /|rsft  | app|  up|pgdn|    |   1|   2|   3| tab|
+     * |-------------------------------------------------------------------------|---------|    |---------------    |
+     * | gui|      |lalt|              space                    |ralt|      |left|down|rght|    |        0|   .|    |
+     * `-----------------------------------------------------------------------------------'    `-------------------'
+     */
+     
+    [0] = KEYMAP(
```
Comment (mechmerlin) [inline keyboards/converter/hp_46010a/keymaps/default/keymap.c]: Replace `KEYMAP` with `LAYOUT`

File: keyboards/converter/hp_46010a/hp_46010a.h [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    {KEY_Y,     KEY_T,    KEY_R,    KEY_E,    KEY_W,    KEY_Q,    KEY_TAB,  KEY_U    }, \
+    {KEY_RES,   KEY_F,    KEY_D,    KEY_S,    KEY_A,    KC_NO,    KEY_2,    KEY_H    }, \
+    {KEY_P7,    KEY_P4,   KEY_P8,   KEY_P5,   KEY_P9,   KEY_P6,   KEY_PADE, KEY_PCOM }, \
+    {KEY_MULT,  KEY_P1,   KEY_DIV,  KEY_P2,   KEY_PLUS, KEY_P3,   KEY_MINU, KEY_PTAB }, \
+    {KEY_B1,    KEY_P0,   KEY_B2,   KC_NO,    KEY_B3,   KEY_PDOT, KEY_B4,   KC_NO    }, \
+    {KEY_B,     KEY_V,    KEY_C,    KEY_X,    KEY_Z,    KEY_PRNT, KEY_DELE, KEY_N    }, \
+}
+
+#define LAYOUT LAYOUT_HP_46010A
+#define KC_LAYOUT_HP_46010A LAYOUT_HP_46010A
```
Comment (noroadsleft) [inline keyboards/converter/hp_46010a/hp_46010a.h]: Line 63 should be deleted.

Comment (listofoptions) [reply keyboards/converter/hp_46010a/hp_46010a.h]: thanks! fixed!

File: keyboards/converter/HP_46010A/config.h (hunk 1 above)
Comment (drashna) [reply keyboards/converter/HP_46010A/config.h]: GCC definitely supports it, and has for a while, now, apparently.

File: keyboards/converter/HP_46010A/readme.md (hunk 3 above)
Comment (drashna) [reply keyboards/converter/HP_46010A/readme.md]: Ideally, yes.  

If needed, dump it into a Gist and like that in the readme, for others to use.

Comment (listofoptions) [reply keyboards/converter/HP_46010A/readme.md]: ok will do!

File: keyboards/converter/HP_46010A/masks.csv (hunk 2 above)
Comment (listofoptions) [reply keyboards/converter/HP_46010A/masks.csv]: see  62d288a, should be updated now

**Comment by drashna** [comment]
Thanks!

**Pull request merged**